src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java

```diff
@@ -361,7 +360,223 @@ private Bitmap doFileInBackground() {
 
     }
 
-    public static boolean cancelPotentialWork(Object file, ImageView imageView) {
+    public static class AvatarGenerationTask extends AsyncTask<Object, Void, Bitmap> {
+        private final WeakReference<ImageView> mImageViewReference;
+        private final WeakReference<MenuItem> mMenuItemReference;
+        private static Account mAccount;
```

Comment:
@tobiasKaminsky , what's the reason to make this ```static```? It's set in the construction of each instance, so, overwritten by the last one created. There will be problems if different avatars are created in parallel.

Could be related to https://github.com/owncloud/android/pull/1559#issuecomment-251933389